BootloaderCommonPkg/Library/PartitionLib/PartitionLib.c

```diff
@@ -388,13 +406,22 @@ FindGptPartitions (
   ReadSize = Gpt->NumberOfPartitionEntries * Gpt->SizeOfPartitionEntry;
   ReadSize = (ReadSize % DevBlockInfo->BlockSize) == 0 ? ReadSize : DevBlockInfo->BlockSize * ((
                ReadSize / DevBlockInfo->BlockSize) + 1);
-  GptEntries = (EFI_PARTITION_ENTRY *) AllocatePool (ReadSize);
-  if (GptEntries == NULL) {
-    Status = EFI_OUT_OF_RESOURCES;
-    goto Done;
+
+  if (mGptEntriesSize < ReadSize) {
```

Comment:
Can we avoid using module global variable for this?   We might have multiple partition instance opened at the same time, it will cause variable sharing between different instance.   Since each instance already has its own private data PART_BLOCK_DEVICE,  the GptEntries can be one member inside this structure.

Reply:
I agree with @mauricema.

Reply:
I am not aware of any scenarios where we will have multiple block devices mounted at the same time (if one boot option fails then we get the next block device's partition info and update the GPT data to the latest block device's entry data).

How to get access to PART_BLOCK_DEVICE data during kernel command line parameter setup?

Reply:
Our code does not do it currently.  But the design allows to open multiple partition handles.  The implementation should consider that.
The partition handle should be available in OsLoader.  It needs to be passed back into the API as a parameter.  And inside the API, from the handle, you can get its private data by typecasting.